partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -1546,13 +1574,27 @@ internal class PartiQLParserDefault : PartiQLParser {
             GeneratedParser.MINUS -> Expr.Binary.Op.MINUS
             GeneratedParser.PERCENT -> Expr.Binary.Op.MODULO
             GeneratedParser.CONCAT -> Expr.Binary.Op.CONCAT
-            GeneratedParser.ANGLE_LEFT -> Expr.Binary.Op.LT
+            GeneratedParser.ANGLE_LEFT -> {
+                if (ctx.stop.type == GeneratedParser.ANGLE_RIGHT) Expr.Binary.Op.NE
+                else Expr.Binary.Op.LT
+            }
             GeneratedParser.LT_EQ -> Expr.Binary.Op.LTE
             GeneratedParser.ANGLE_RIGHT -> Expr.Binary.Op.GT
             GeneratedParser.GT_EQ -> Expr.Binary.Op.GTE
-            GeneratedParser.NEQ -> Expr.Binary.Op.NE
+            GeneratedParser.BANG -> Expr.Binary.Op.NE
             GeneratedParser.EQ -> Expr.Binary.Op.EQ
-            else -> throw error(token, "Invalid binary operator")
+            else -> throw error(ctx.start, "Invalid binary operator")
+        }
+
+        private fun convertBinaryMathOp(token: Token) = when (token.type) {
+            GeneratedParser.AMPERSAND -> Expr.Binary.Op.BITWISE_AND
+            GeneratedParser.CONCAT -> Expr.Binary.Op.CONCAT
+            GeneratedParser.PLUS -> Expr.Binary.Op.PLUS
+            GeneratedParser.MINUS -> Expr.Binary.Op.MINUS
+            GeneratedParser.PERCENT -> Expr.Binary.Op.MODULO
+            GeneratedParser.ASTERISK -> Expr.Binary.Op.TIMES
+            GeneratedParser.SLASH_FORWARD -> Expr.Binary.Op.DIVIDE
+            else -> throw error(token, "Invalid binary math operator")
```

Comment:
nit: error message can be confusing since `AMPERSAND` and `CONCAT` aren't math operators. Could consider reusing the `convertBinaryOp` error message.

```suggestion
            else -> throw error(token, "Invalid binary operator")
```

Reply:
Also naming for the function can be confusing (not that big an issue since it's private). Could just use the name `convertBinaryOp` since the first argument is different. And add a comment that this is used for `mathOp00`, `mathOp01`, and `mathOp02`.